spec/requests/requests_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe 'managing topics', :type => :request do
+
+  let!(:user) {FactoryBot.create :user}
+
+  it 'redirects unauthenticated users' do
+    post "/topics", :params => { :topic => {:name => "here is a new topic"} }
+    expect(response).to redirect_to('/users/sign_in')
+  end
+
+  it 'allows topic posting for logged in users' do
+    post user_session_path, :params => {'user[email]' => user.email, 'user[password]' => user.password }
+    get '/topics/new'
+    expect(response).to render_template(:new)
+
+    post "/topics", :params => { :topic => {:name => "here is a new topic"} }
```

Comment:
Don't test multiple requests in one spec (e.g in this test your testing a get#new and a post#create request).
For the test structure I would do a describe for each controller action.

```ruby
describe 'put#update' do
  it 'is forbidden for logged out users' do
    ...
  end
  
  it 'is allowed for authenticated users' do
    ...
  end
end